pkg/params/params.go

```diff
@@ -352,6 +352,53 @@ func (p *Param) Set(val string) error {
 	return nil
 }
 
+func (p *Param) IsSet() bool {
+	return p.DefaultValue != p.value
+}
```

Comment:
How is `IsSet()` used? Maybe it's better to add a boolean that is set when `Set()` is called and check for that - and then add another func `IsDefault() bool` that does this comparison (negated).

Reply:
TBH I don't remember what was the reason I chose this implementation. I think it was related to `Set()` being called in by the service (while decoding params from the client), and that causing some issues with the default value of some ebpf params. I tried to add a boolean to param, tested with https://github.com/inspektor-gadget/inspektor-gadget/pull/2119  and worked fine, so I'll stick to that solution. I also added the `IsDefault()` method as it could be useful in other situations.

Reply:
Awesome, thanks for adding that!